### ISSUE TITLE
Request to add DropEdge

### ISSUE THREAD

**Issue by AnirudhDagar** [rusty1s/pytorch_geometric#1115]
## 🚀 Feature

Request to add **DropEdge: Towards Deep Graph Convolutional Networks on Node Classification**, (ICLR 2020) to Pytorch Geometric.

Paper: https://arxiv.org/pdf/1907.10903.pdf
PyTorch Implementation: https://github.com/DropEdge/DropEdge

## Motivation

This paper proposes DropEdge, to tackle the over-smoothing which can hinder model training in DeepGCNs and also has effects of better generalization much like dropout.

Cheers!

**Comment by rusty1s** [comment]
That should be already implemented via [`dropout_adj`](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.dropout_adj)?

**Comment by AnirudhDagar** [comment]
Great! Had to run some experiments.
